cmrtlib/linux/CMakeLists.txt

```diff
@@ -90,8 +91,26 @@ endif()
 set_target_properties( igfxcmrt PROPERTIES 
   FOLDER CM_RT 
   COMPILE_DEFINITIONS "${CMRT_DEFINES}" 
-  OUTPUT_NAME "igfxcmrt${TARGET_MODIFIER}"
+  OUTPUT_NAME "libigfxcmrt${TARGET_MODIFIER}"
   )
+
+# set the version, aligned with media driver
+if(NOT DEFINED MEDIA_VERSION OR MEDIA_VERSION STREQUAL "")
+    set(MEDIA_VERSION "1.0.0")
+endif()
+set(CMRT_VERSION ${MEDIA_VERSION})
+string(REPLACE "." ";" CMRT_VERSION_LIST ${CMRT_VERSION})
+list(GET CMRT_VERSION_LIST 0 CMRT_VERSION_MAJOR)
+
+set_target_properties(igfxcmrt PROPERTIES VERSION ${CMRT_VERSION})
+set_target_properties(igfxcmrt PROPERTIES SOVERSION ${CMRT_VERSION_MAJOR})
 set_target_properties( igfxcmrt PROPERTIES PREFIX "")
 target_link_libraries( igfxcmrt dl va rt ${GCC_SECURE_LINK_FLAGS})
 
+ADD_CUSTOM_TARGET(link_target ALL COMMAND ${CMAKE_COMMAND} -E create_symlink libigfxcmrt${TARGET_MODIFIER}.so igfxcmrt${TARGET_MODIFIER}.so)
```

Comment:
This is useless command. This should be automatically handled by cmake with install() commands below (the one with NAMELINK_ONLY). For that user needs to execute "make install".

Reply:
No, you don't. You don't actually have backward compatibility point in any of Linux distributions and don't pretend that you have. All what was before was custom patches over existing linux distributions, they don't count. You have nice opportunity to simplify the things and make them right. I strongly suggest to use it.

Reply:
For those applications who already depends on igfxcmrt64.so, obviously this will block them. They will find they can't find hte igfxcmrt64.so when they update their media driver

Reply:
you let other applications know that they need to adjust by bumping the ABI/API version.  This is exactly what should be fixed for good.

Reply:
yes, i have dropped the old igfxcmrt64.so in last update. please review the latest files change